Add ctrl+click and alt+click shortcuts for cryo, as well as a verb to remove beakers.

ShiftyRail opened the pull request:
Reasoning : you can remove the patient with both the verb and nanoUI, why not do the same with the beaker ?
And while we're at it, why not add some shortcuts to have a better _quality of life_ ?
(oh and I removed a trailing return and a typo error in passing)
Tested.
Fixes #16342
:cl:
- tweak: You can now remove the beaker from a cryotube with a verb accessible in the right-click contextual menu.
- rscadd: Adds shortcuts to the cryocell. Alt+click to remove the patient within it (including yourself), Ctrl+click to remove the beaker.

DamianX commented on code/game/machinery/cryo.dm:
```diff
@@ -637,6 +637,20 @@ var/global/list/cryo_health_indicator = list(	"full" = image("icon" = 'icons/obj
+	set name = "Remove beaker"
+	set category = "Object"
+	set src in oview(1)
+	if(usr.incapacitated() || usr.lying || usr.locked_to || !(iscarbon(usr) || issilicon(usr))) //are you cuffed, dying, lying, stunned or other
```
` !(iscarbon(usr) || issilicon(usr))` are you sure about this?

ShiftyRail replied:
I tested with a human (worked), a silicon (worked), a mouse (didn't) and a ghost (didn't).
I imagine we don't want mouses removing beakers, and stuff like that.

DamianX commented on code/game/machinery/cryo.dm:
```diff
@@ -637,6 +637,20 @@ var/global/list/cryo_health_indicator = list(	"full" = image("icon" = 'icons/obj
+	if(beaker)// If there is, effectively, a beaker
+		detach()
+	add_fingerprint(usr)
+	return
```
return at the end of the proc does nothing

ShiftyRail replied:
Whoops
To be fair, I was copying the structure of the proc above which also has a trailing return. Should I remove it as well ?

SonixApache commented:
why not just fix the nanoUI instead

PJB3005 commented:
@SonixApache the UI is *supposed* to not open when the machine is depowered.

ShiftyRail commented:
@PJB3005 this might be worth checking, because nanoUI does open, it just doesn't register your clicks. 🤔

PJB3005 commented:
Oh yeah that's unintentional then.

Shadowmech88 commented:
Do cryo tubes already have Alt+click and/or Ctrl+click functionality?
Since verbs are suboptimal, maybe the beaker removal could be done via one of those?

ShiftyRail commented:
I'll try that. Both options would be available.
Alt+click for the patient, ctrl+click for beaker.

In the meantime, don't merge this yet.

ShiftyRail commented:
Changed it in a rather big way. Would appreciate some feedback again, please.

ShiftyRail commented on code/game/machinery/cryo.dm:
```diff
@@ -601,26 +601,10 @@ var/global/list/cryo_health_indicator = list(	"full" = image("icon" = 'icons/obj
-		go_out()
-	add_fingerprint(usr)
-	return
+	AltClick(usr)
```
Don't really know if `user` would work here.

DamianX replied:
it will if you define it as argument for the proc

DamianX commented on code/game/machinery/cryo.dm:
```diff
@@ -666,6 +642,35 @@ var/global/list/cryo_health_indicator = list(	"full" = image("icon" = 'icons/obj
 		message_admins("[key_name(L)] has ejected [occupant] from \the [src]! [formatJumpTo(src)]")
 		go_out()
 
+/obj/machinery/atmospherics/unary/cryo_cell/AltClick(mob/user as mob) // AltClick = most common action = removing the patient
```
remove `as mob`

ShiftyRail commented:
Removed the `as mob` (too bad, I liked them), and also added a check whether or not the user is the cryocell to remove the beaker. The idea is to avoid your user removing your beaker with 50u polyacid.
